pkg/agent/client/dial.go

```diff
@@ -65,9 +71,17 @@ func DialServer(ctx context.Context, config DialServerConfig) (*grpc.ClientConn,
 		}
 	}
 
-	return grpc.DialContext(ctx, config.Address,
+	ctx, cancel := context.WithTimeout(ctx, _defaultDialTimeout)
+	defer cancel()
+
+	client, err := grpc.DialContext(ctx, config.Address,
 		grpc.WithBalancerName(roundrobin.Name), //nolint:staticcheck
 		grpc.FailOnNonTempDialError(true),
+		grpc.WithBlock(),
```

Comment:
`WithBlock()` is required here: `FailOnNonTempDialError only affects the initial dial, and does not do anything useful unless you are also using WithBlock().` (from https://godoc.org/google.golang.org/grpc#FailOnNonTempDialError)